pricelist_item_generator/models/pricelist.py

```diff
@@ -0,0 +1,415 @@
+# coding: utf-8
+# © 2016 David BEAL @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import fields, models, api, _
+from openerp.exceptions import Warning as UserError
+import openerp.addons.decimal_precision as dp
+from collections import defaultdict
+
+
+TODO_SELECTION = [('update', 'Update'), ('create', 'Create')]
+
+
+def update_item_tpl_or_condition(vals, self):
```

Comment:
IMO it would make more sense if `self` is the first arg in order to match standard Python instance methods.

Also the `api.multi` decorator should be added to this method in order to be more compatible.